framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -624,19 +614,29 @@ private void replaceAnnotations(AnnotatedTypeMirror destType, AnnotatedTypeMirro
      * @param e element whose outermost enclosing class will be parsed
      */
     private void parseEnclosingClass(Element e) {
-        if (!shouldParseJdk) {
+        if (!shouldParseJdk
+                || e.getKind() == ElementKind.PACKAGE
+                || e.getKind() == ElementKind.MODULE) {
             return;
         }
+
         String className = getOutermostEnclosingClass(e);
-        if (className == null || className.isEmpty()) {
+        if (!processingClasses.add(className)) {
```

Comment:
It seems some test cases are failing after adding the assertion.

Reply:
The parent of `Array` is an element of kind `OTHER`:

<img width="710" alt="image" src="https://user-images.githubusercontent.com/12403206/180348185-256eec9c-08fe-437d-a6ff-1d53496875b0.png">

and it doesn't have an enclosing class. This issue can be caused by having `Object[]` in the source code.

I think we need to add the null and empty check back.

Reply:
Wouldn't that mean that declaration annotations don't influence array types, because the enclosing class wouldn't be parsed?
Do you have a minimized test case that reproduces this?
Is this something that should be fixed in `getOutermostEnclosingClass`?

Reply:
But is there a way to add declaration annotations to array types? There's no `Array.java` class (the one under `reflect` is different), and the name of its enclosing package is empty. It looks like a pseudo type element to me.

Reply:
Ah. So should Array be added to the first `if` in this method (together with Package and Module) to exclude that case? Then you can have the assert here and make sure that if some other unexpected thing shows up, we notice from the assert failure.

Reply:
added a TODO for future investigation